bin/check_h5_plugin.cpp

```diff
@@ -33,35 +92,54 @@ int main(int argc, char* argv[]) {
         return -1;
     }
     std::cerr << "[ OK ] plugin_get_header successful\n";
-    std::cerr << "   nx      " << nx << "\n";
-    std::cerr << "   ny      " << ny << "\n";
-    std::cerr << "   nbytes  " << nbytes << "\n";
-    std::cerr << "   nframes " << nframes << "\n";
-    std::cerr << "   qx      " << qx << "\n";
-    std::cerr << "   qy      " << qy << "\n";
+    std::cerr << "    nx      " << nx << "\n";
+    std::cerr << "    ny      " << ny << "\n";
+    std::cerr << "    nbytes  " << nbytes << "\n";
+    std::cerr << "    nframes " << nframes << "\n";
+    std::cerr << "    qx      " << qx << "\n";
+    std::cerr << "    qy      " << qy << "\n";
+    if (nframes <= 0) {
+        std::cerr << "[FAIL] nframes must be positive\n";
+        return -1;
+    }
     if (qx <= 0.0 || qy <= 0.0) {
         std::cerr << "[WARN] qx/qy should be positive (pixel sizes) - either "
                      "missing or wrong\n";
     }
 
     auto dataArrayExtracted = std::unique_ptr<int[]>(new int[nx * ny]);
-
+    std::cerr << "  trying to extract all data frames. please wait...\n";
+    std::cerr << "    [" << std::setw(6) << std::right << "1"
+              << " / " << nframes << " ]\n";
     for (int frame = 1; frame <= nframes; ++frame) {
-        plugin_get_data(&frame, &nx, &ny, dataArrayExtracted.get(), info_array,
-                        &error_flag);
+        if ((frame % 100) == 0) {
+            std::cerr << "    [" << std::setw(6) << std::right << frame << " / "
+                      << nframes << " ]\n";
+        }
+        get_data(&frame, &nx, &ny, dataArrayExtracted.get(), info_array,
+                 &error_flag);
         if (error_flag != 0) {
             std::cerr << "[FAIL] plugin_get_data for frame " << frame
                       << " returned error " << error_flag << "\n";
             return -1;
         }
     }
 
-    plugin_close(&error_flag);
+    close_file(&error_flag);
     if (error_flag != 0) {
         std::cerr << "[FAIL] plugin_close returned error " << error_flag
                   << "\n";
         return -1;
     }
     std::cerr << "[ OK ] plugin_close successful\n";
+
+    auto end_time = std::chrono::system_clock::now();
+    std::cerr << "\n==== " << time_string(end_time) << " ====\n";
+    std::cerr << "\nelapsed time: "
+              << std::chrono::duration_cast<std::chrono::milliseconds>(
+                         end_time - start_time)
+                                 .count() /
+                         1000.0
+              << " seconds\n";
```

Comment:
I suppose the user only cares about the elapsed time here. However, I don't think this tool needs to have this responsibility at all. Starting this program with the `time` command is adequate since you are only measuring start to end.

Reply:
removed all logs related to timings